src/ATen/dlconvertor.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "ATen/Tensor.h"
+#include "ATen/ATen.h"
+#include "ATen/dlpack.h"
+
+// this convertor will:
+// 1) take a Tensor object and wrap it in the DLPack tensor object
+// 2) take a dlpack tensor and convert it to the Tensor object
+
+namespace at { namespace dlpack {
+
+using DLTensorSPtr = std::shared_ptr<DLTensor>;
+
+class DLConvertor {
```

Comment:
There is no need for this to be a class. Just have:

```
void toDLPack(at::Tensor src, DLTensor * dst);
at::Tensor fromDLPack(DLTensor * src);
```

Everything else can be static functions in the `.cpp` file.